### PR TITLE
Added equality checks to ActiveMenu

### PR DESCRIPTION
**Pull request by FuJa0815**
**Brief Description of What This PR Does**

This PR stops the user from repeadatly pressing F1 to randomize the easter egg message.

**Progress Checklist**

Note: before starting this checklist the issue should be marked as non-draft.

- [ ] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

**Comment by Kasterisk** [comment]
Personally I kind of like being able to randomize the easter eggs on demand but that would defeat their purpose I guess :P.

Anyways a better way for this is to have an enum specifying the menus similarly to how the tabs in MicrobeEditorGUI are handled. So SetActiveMenu method could be left as is and the enum could be used to check what's the current opened menu. See [MicrobeEditorGUI.cs#L1474](https://github.com/Revolutionary-Games/Thrive/blob/master/src/microbe_stage/editor/MicrobeEditorGUI.cs#L1474) and [MicrobeEditorGUI.cs#L409](https://github.com/Revolutionary-Games/Thrive/blob/master/src/microbe_stage/editor/MicrobeEditorGUI.cs#L409)

**Comment by FuJa0815** [comment]
I was thinking about this too. I was wondering why this was implemented with strings and not enums. I'll change this to an enum-based approach.

**Comment by Kasterisk** [comment]
The PauseMenu class predates the enum based tab/page handling in the codebase so it just needs some a bit of refactoring.

File: src/general/PauseMenu.cs [hunk 1]
```diff
@@ -49,11 +49,55 @@ public class PauseMenu : ControlWithInput
     [Signal]
     public delegate void MakeSave(string name);
 
+    private enum MenuEnum
```
Comment (hhyyrylainen) [inline src/general/PauseMenu.cs:52]:
```suggestion
    private enum ActiveMenuType
```
I don't like using Enum as a suffix...

File: src/general/PauseMenu.cs [hunk 2]
```diff
@@ -49,11 +49,55 @@ public class PauseMenu : ControlWithInput
...
+    }
+
     /// <summary>
     ///   The GameProperties object holding settings and state for the current game session.
     /// </summary>
     public GameProperties GameProperties { get; set; }
 
+    private MenuEnum ActiveMenu
+    {
+        get
```
Comment (hhyyrylainen) [inline src/general/PauseMenu.cs:69]: I hope this get isn't called very often...

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Other than that one naming thing this looks good.

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**